kalastatic.module

```diff
@@ -126,11 +133,12 @@ function kalastatic_get_settings($type = 'yaml', $adjust_path_roots = TRUE) {
       // Also refresh the config. Note that this relies on the data in
       // kalastatic.yaml so we need to make sure we have that first or we'll get
       // stuck in a loop.
-      $settings = kalastatic_get_kalastatic_config($settings);
+      $yaml = kalastatic_adjust_path_root($data['yaml']);
+      $data['config'] = kalastatic_get_kalastatic_config($yaml);
```

Comment:
Doh, I just realized we're never actually _using_ the cached `config` data. We set config in the data cache, but every time it's requested, we just fetch it manually again.

I say we just warm `config` up there in the `yaml` cache miss `else` statement. And in this switch case, we just need the one `$settings = $data['config'];` line. Going further, if you remove the `$adjust_path_roots` parameter (optional), then we can do away with the switch statement entirely and just:
```
return $data[$type];
```

Finally, we can change all the cache warming calls back to just `kalastatic_get_settings()` (i.e., no `'config'` parameter). Round and round we go...